core/src/main/java/org/bitcoinj/store/WalletProtobufSerializer.java

```diff
@@ -450,33 +491,40 @@ public Wallet readWallet(NetworkParameters params, @Nullable WalletExtension[] e
             wallet.setDescription(walletProto.getDescription());
         }
 
-        // Read all transactions and insert into the txMap.
-        for (Protos.Transaction txProto : walletProto.getTransactionList()) {
-            readTransaction(txProto, wallet.getParams());
-        }
-
-        // Update transaction outputs to point to inputs that spend them
-        for (Protos.Transaction txProto : walletProto.getTransactionList()) {
-            WalletTransaction wtx = connectTransactionOutputs(txProto);
-            wallet.addWalletTransaction(wtx);
-        }
-
-        // Update the lastBlockSeenHash.
-        if (!walletProto.hasLastSeenBlockHash()) {
+        if (forceReset) {
+            // Should mirror Wallet.reset()
```

Comment:
That being said, someone editing `Wallet.reset` may not know to edit this as well. I can refactor the common code out to a `private` or `protected` method if you think it's worth it?